### PR TITLE
Gitlab notifications

### PR DESCRIPTION
**Pull request by kpacha**
send build status updates to gitlab as comments on the related commit

### CODE REVIEW

File: plugin/notify/gitlab.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+type GitlabClient interface {
+	SendRepoCommitComment(id string, sha string, body string) (*gogitlab.CommitComment, error)
+}
+
+const (
+	StatusPending = ":raised_hand:"
+	StatusSuccess = ":thumbsup:"
+	StatusFailure = ":thumbsdown:"
+	StatusError   = ":exclamation:"
```
Comment (chengweiv5) [inline plugin/notify/gitlab.go]: Add 'Started/Running' status?

**Comment by kpacha** [comment]
@chengweiv5: the Started/Running status is not implemented because this plugin doesn't update the gitlab CI_Server status (https://gitlab.com/gitlab-org/gitlab-ce/blob/master/app/models/project_services/ci_service.rb#L44). It just sends a comment and I think it would be too verbose.

For an actual integration, I'd rather build a new gitlab CI_Service.

**Comment by partikus** [comment]
:+1:

**Comment by bradrydzewski** [comment]
cool, merging this now, but just a heads up that 0.4-alpha.1 is going to release next week with a new plugin model and this will need to be manually ported over

**Pull request merged**